From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VerteDinde <vertedinde@electronjs.org>
Date: Tue, 1 Mar 2022 12:07:25 -0800
Subject: build: disable partition alloc on mac

Enabling partition alloc caused a crash when spawning
a child process. This patch disables partition alloc for mac,
and can be removed when the crash in fork is resolved.
Related issue: https://github.com/electron/electron/issues/32718

diff --git a/build_overrides/partition_alloc.gni b/build_overrides/partition_alloc.gni
index e90ccba879dc5823c15acf35869cd5bd211b717b..a95d2b46b37e007405cd472bd9f882c1d180d809 100644
--- a/build_overrides/partition_alloc.gni
+++ b/build_overrides/partition_alloc.gni
@@ -46,7 +46,7 @@ _disable_partition_alloc_everywhere = is_component_build || (is_win && is_debug)
 if (is_ios) {
   _is_partition_alloc_everywhere_platform = ios_partition_alloc_enabled
 } else {
-  _is_partition_alloc_everywhere_platform = !is_nacl
+  _is_partition_alloc_everywhere_platform = !is_nacl && !is_mac
 }
 
 # Under Windows debug build, the allocator shim is not compatible with CRT.
